### DISCUSSION TITLE
Include email transport mailjet

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#19450] [Feature Requests]
### Summary

In node_modules/@directus/api/dist/mailer.js could we have
```
    else if (transportName === 'mailjet') {
	const mg = require('nodemailer-smtp-transport'); // npm i nodemailer-smtp-transport
        transporter = nodemailer.createTransport(mg({
            auth: {
                api_key: env['MAILJET_API_KEY'],
                domain: env['MAILJET_API_SECRET'],
            },
            host: 'in-v3.mailjet.com',
        }));
    }
```
while in .env
```
## Email (Mailjet Transport)
MAILJET_API_KEY="<mailjet_api_key>"
MAILJET_API_SECRET="<mailjet_api_secret>"

```

### Basic Example

see above

### Motivation

Mailjet (acquired by mailgun which then was acquired by Sinch) has a solid audience.

### Detailed Design

see above

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

-

### Alternatives

-

### Adoption Strategy

-

**Comment by rijkvanzanten** [comment]
It sounds to me like you can already use mailjet with the existing setup

https://github.com/directus/directus/blob/dbd56a40edd9a5d59a7461e458c04cd14e8cb730/api/src/mailer.ts#L61-L65

With your example values:

```
EMAIL_MAILGUN_API_KEY=""<mailjet_api_key>""
EMAIL_MAILGUN_DOMAIN="<mailjet_api_secret>"
EMAIL_MAILGUN_HOST="in-v3.mailjet.com"
```

I don't really see any reason to add additional environment variables if these already do what you're asking for 🙂

**Reply by rijkvanzanten** [reply]
Also please fill out the whole form, the sections are there to be able to properly discuss implementation details and are crucial!

**Comment by Drunkenpilot** [comment]
> It sounds to me like you can already use mailjet with the existing setup
> 
> https://github.com/directus/directus/blob/dbd56a40edd9a5d59a7461e458c04cd14e8cb730/api/src/mailer.ts#L61-L65
> 
> With your example values:
> 
> ```
> EMAIL_MAILGUN_API_KEY=""<mailjet_api_key>""
> EMAIL_MAILGUN_DOMAIN="<mailjet_api_secret>"
> EMAIL_MAILGUN_HOST="in-v3.mailjet.com"
> ```
> 
> I don't really see any reason to add additional environment variables if these already do what you're asking for 🙂

Unfortunately, it didn't worked.
We got an error after a tried with flows:
```
{
  "name": "Error",
  "message": "HPE_INVALID_CONSTANT"
}
```

So Could it be possible to add mailjet support ? It'll be really helpful. Thanks 😊

**Reply by Vincent-HD** [reply]
Any news on this ?

**Reply by Drunkenpilot** [reply]
Finally we finished setup with SMTP by using Mailjet SMTP Relay
```
### Email
EMAIL_FROM:mailjet_validated_sender_address
EMAIL_TRANSPORT:"smtp"

### Email SMTP
EMAIL_SMTP_HOST:"in-v3.mailjet.com"
EMAIL_SMTP_PORT:"465"
EMAIL_SMTP_USER:mailjet_api_key
EMAIL_SMTP_PASSWORD:mailjet_api_secret
EMAIL_SMTP_SECURE: true
```

**Reply by Vincent-HD** [reply]
Hmm, I'll try that, I think I did yesterday through the SMTP relay, but it seems that it didn't work. Thx for sharing your config tho it'll be useful 👍

**Reply by Drunkenpilot** [reply]
You're welcome. Let me know whether it works or not.

**Reply by Vincent-HD** [reply]
It works 💯 It was related to one of my account settings. thx

**Reply by Drunkenpilot** [reply]
Great to hear that ! :)
